WorkloadManagementSystem/Agent/JobAgent.py

```diff
@@ -345,283 +346,283 @@ def execute( self ):
 
   #############################################################################
-  def __report( self, jobID, status, minorStatus ):
+  def __report(self, jobID, status, minorStatus):
     """Wraps around setJobStatus of state update client
     """
-    jobReport = RPCClient( 'WorkloadManagement/JobStateUpdate' )
-    jobStatus = jobReport.setJobStatus( int( jobID ), status, minorStatus, 'JobAgent@%s' % self.siteName )
-    self.log.verbose( 'setJobStatus(%s,%s,%s,%s)' % ( jobID, status, minorStatus, 'JobAgent@%s' % self.siteName ) )
+    jobReport = JobStateUpdateClient()
+    jobStatus = jobReport.setJobStatus(int(jobID), status, minorStatus, 'JobAgent@%s' % self.siteName)
+    self.log.verbose('setJobStatus(%s,%s,%s,%s)' % (jobID, status, minorStatus, 'JobAgent@%s' % self.siteName))
     if not jobStatus['OK']:
-      self.log.warn( jobStatus['Message'] )
+      self.log.warn(jobStatus['Message'])
 
     return jobStatus
 
   #############################################################################
-  def __setJobParam( self, jobID, name, value ):
+  def __setJobParam(self, jobID, name, value):
     """Wraps around setJobParameter of state update client
     """
-    jobReport = RPCClient( 'WorkloadManagement/JobStateUpdate' )
-    jobParam = jobReport.setJobParameter( int( jobID ), str( name ), str( value ) )
-    self.log.verbose( 'setJobParameter(%s,%s,%s)' % ( jobID, name, value ) )
+    jobReport = JobStateUpdateClient()
+    jobParam = jobReport.setJobParameter(int(jobID), str(name), str(value))
+    self.log.verbose('setJobParameter(%s,%s,%s)' % (jobID, name, value))
     if not jobParam['OK']:
-      self.log.warn( jobParam['Message'] )
+      self.log.warn(jobParam['Message'])
 
     return jobParam
 
   #############################################################################
-  def __finish( self, message, stop = True ):
+  def __finish(self, message, stop=True):
     """Force the JobAgent to complete gracefully.
     """
     if stop:
-      self.log.info( 'JobAgent will stop with message "%s", execution complete.' % message )
+      self.log.info('JobAgent will stop with message "%s", execution complete.' % message)
       self.am_stopExecution()
-      return S_ERROR( message )
+      return S_ERROR(message)
     else:
-      return S_OK( message )
+      return S_OK(message)
 
   #############################################################################
-  def __rescheduleFailedJob( self, jobID, message, stop = True ):
+  def __rescheduleFailedJob(self, jobID, message, stop=True):
     """
     Set Job Status to "Rescheduled" and issue a reschedule command to the Job Manager
     """
 
-    self.log.warn( 'Failure during %s' % ( message ) )
+    self.log.warn('Failure during %s' % (message))
 
-    jobManager = RPCClient( 'WorkloadManagement/JobManager' )
-    jobReport = JobReport( int( jobID ), 'JobAgent@%s' % self.siteName )
+    jobManager = JobManagerClient()
```

Comment:
I think we can simply define 
```self.jobManager = JobManagerClient()```
in the `initialize` and avoid repeating the same code all over.